packages/aws-rfdk/lib/deadline/lib/database-connection.ts

```diff
@@ -267,6 +276,35 @@ class DocDBDatabaseConnection extends DatabaseConnection {
     }
   }
 
+  /**
+   * @inheritdoc
+   */
+  public addSecurityGroup(...securityGroups: ISecurityGroup[]): void {
+    let added = false;
+    if (this.props.database instanceof DatabaseCluster) {
+      const resource = (this.props.database as DatabaseCluster).node.findChild('Resource');
+      if (resource instanceof CfnDBCluster) {
+        const cfnCluster = resource as CfnDBCluster;
+        const securityGroupIds = securityGroups.map(sg => sg.securityGroupId);
+
+        if (cfnCluster.vpcSecurityGroupIds === undefined) {
+          cfnCluster.vpcSecurityGroupIds = securityGroupIds;
+        } else {
+          cfnCluster.vpcSecurityGroupIds.push(...securityGroupIds);
+        }
+        added = true;
+      }
+    }
+
+    if (!added) {
+      Annotations.of(this.props.database).addWarning(
+        `Failed to add the following security groups to ${this.props.database.node.id}: ${securityGroups.map(sg => sg.node.id).join(', ')}. ` +
+        'This is because either the "database" property passed to this class is not an instance of AWS CDK\'s DocumentDB cluster construct or the ' +
+        'internal implementation of AWS CDK\'s DocumentDB cluster construct has changed.',
+      );
+    }
```

Comment:
What do you say on adding separate annotation far every  case instead of general one?
I believe it will be more informative for user.